website/src/app/layout.tsx

```diff
@@ -0,0 +1,91 @@
+import { FC, ReactNode } from 'react';
+import { GitHubIcon, PaperIcon, PencilIcon, PRODUCTS, VersionDropdown } from '@theguild/components';
+import { getDefaultMetadata, getPageMap, GuildLayout } from '@theguild/components/server';
+import '@theguild/components/style.css';
+import { pageMap as changelogsPageMap } from './changelogs/[...slug]/page';
+import { pageMap as v2PageMap } from './v2/[[...slug]]/page';
+import { pageMap as v3PageMap } from './v3/[[...slug]]/page';
+import { pageMap as v4PageMap } from './v4/[[...slug]]/page';
+import { VersionedSearch } from './versioned-search';
+
+const description = PRODUCTS.YOGA.title;
+const websiteName = 'Yoga';
+
+export const metadata = getDefaultMetadata({
+  description,
+  websiteName,
+  productName: 'YOGA',
+});
+
+const RootLayout: FC<{
+  children: ReactNode;
+}> = async ({ children }) => {
+  let [meta, ...pageMap] = await getPageMap();
+  pageMap = [
+    {
+      data: {
+        // @ts-expect-error -- ignore
+        ...meta.data,
+        changelogs: { type: 'page', title: 'Changelogs', theme: { layout: 'full' } },
+        v2: { type: 'page', title: 'Yoga 2 Docs' },
+        v3: { type: 'page', title: 'Yoga 3 Docs' },
+        v4: { type: 'page', title: 'Yoga 4 Docs' },
+      },
+    },
+    ...pageMap,
+    { route: '/changelogs', name: 'changelogs', children: changelogsPageMap },
+    { route: '/v4', name: 'v4', children: v4PageMap },
+    { route: '/v3', name: 'v3', children: v3PageMap },
+    { route: '/v2', name: 'v2', children: v2PageMap },
+  ];
```

Comment:
Out of curiosity: why do we need to change the page map for versions here?
Won't Nextra automatically know about these directories?

Reply:
nextra doesn't create `pageMap` for dynamic folders for now